bazel/foreign_cc/BUILD

```diff
@@ -0,0 +1,83 @@
+licenses(["notice"])  # Apache 2
+
+load("//bazel:envoy_build_system.bzl", "envoy_package")
+load("@rules_foreign_cc//tools/build_defs:cmake.bzl", "cmake_external")
+
+envoy_package()
+
+cmake_external(
+    name = "ares",
+    cache_entries = {
+        "CARES_SHARED": "no",
+        "CARES_STATIC": "on",
+        "CMAKE_BUILD_TYPE": "RelWithDebInfo",
+    },
+    cmake_options = ["-GNinja"],
+    lib_source = "@com_github_c_ares_c_ares//:all",
+    make_commands = [
+        "ninja",
+        "ninja install",
```

Comment:
Do we actually need the `ninja install` step?

Reply:
Yes, but previously `ninja install` was used to move ("install") build artifacts into `$THIRDPARTY_BUILD` (which we don't specify here, btw.), which this PR eliminates, so it's integral part of this change, and not a random fix-up, IMHO.

I honestly don't know whether this is still needed or not with `cmake_external`, but it could be easily tested by removing the `ninja install` step.

(This applies to all dependencies, not only this one).